images/devtools-kubernetes-v1beta1/context/magefiles/metadata.go

```diff
@@ -0,0 +1,46 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+)
+
+type Annotations struct {
+	ProjectSlug string `yaml:"github.com/project-slug"`
+}
+
+type Metadata struct {
+	Annotations Annotations `yaml:"annotations"`
+}
+
+type CatalogInfo struct {
+	Metadata Metadata `yaml:"metadata"`
+}
+
+func repoName() (string, error) {
+	var catalogInfo CatalogInfo
+
+	yamlFile, err := ioutil.ReadFile("catalog-info.yaml")
+	if err != nil {
+		fmt.Printf("yamlFile.Get err #%v ", err)
+		return "", err
+	}
+	err = yaml.Unmarshal(yamlFile, &catalogInfo)
+	if err != nil {
+		fmt.Printf("Unmarshal: %v", err)
+		return "", err
+	}
+	return catalogInfo.Metadata.Annotations.ProjectSlug, nil
```

Comment:
Perhaps not something specific for this PR, but something I noticed when playing around with this was that debugging the "recipes" required me to add some logging. Some of our existing Makefiles support logging more information if you specify `VERBOSE=all` or similar. One can of course specify `-v` to mage-k8s to get a little more output, but it doesn't seem to provide that much more.

I might be missing something but perhaps at some point we can consider wrapping some worthy vars in a log-if-debug style wrapper.

Reply:
we indeed have to find the best way here to run (debug) logs here. There are multiple places where you can do this. There is also a `-debug` switch. A question also, instead of fmt, maybe use a log library. I am not sure yet what the best approach here is.